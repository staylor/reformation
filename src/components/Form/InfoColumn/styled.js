import { css } from 'pretty-lights';
import themeUtils from 'styles/theme';

export const infoColumnClass = css`
  float: right;
  line-height: 1.4;
  margin: 10px -300px 0 0;
  position: sticky;
  top: 10px;
  width: 280px;

  @media screen and (max-width: 782px) {
    float: none;
    margin-right: 0;
    width: 100%;
  }
`;

export const infoBoxClass = css`
  border: 1px solid ${themeUtils.colors.detail};
  box-shadow: 0 1px 1px rgba(0, 0, 0, 0.04);
  box-sizing: border-box;
  display: block;
  margin: 0 0 20px;
`;

export const infoBoxHeaderClass = css`
  border-bottom: 1px solid ${themeUtils.colors.detail};
  color: ${themeUtils.colors.dark};
  font-size: 14px;
  font-weight: ${themeUtils.fonts.weight.bold};
  padding: 8px 12px;
  user-select: none;
`;

export const infoBoxContentClass = css`
  font-size: 13px;
  padding: 6px 10px 20px;
`;
