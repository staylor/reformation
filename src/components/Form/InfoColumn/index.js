import React from 'react';
import { infoColumnClass, infoBoxClass, infoBoxHeaderClass, infoBoxContentClass } from './styled';

// type Props = {
//   infoFields?: Array<any>,
//   metaFields?: Array<any>,
//   label?: string,
//   button?: any,
// };

/* eslint-disable react/prop-types */

export default function InfoColumn({ infoFields, metaFields, label, button }) {
  return (
    <section className={infoColumnClass}>
      {infoFields.length > 0 ? (
        <aside className={infoBoxClass}>
          <h3 className={infoBoxHeaderClass}>{label}</h3>
          <div className={infoBoxContentClass}>
            {infoFields}
            {button}
          </div>
        </aside>
      ) : null}
      {metaFields.length > 0 ? (
        <aside className={infoBoxClass}>
          <div className={infoBoxContentClass}>{metaFields}</div>
        </aside>
      ) : null}
    </section>
  );
}

InfoColumn.defaultProps = {
  infoFields: [],
  metaFields: [],
  label: '',
  button: null,
};
