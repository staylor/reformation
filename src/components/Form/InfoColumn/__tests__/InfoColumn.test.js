import React from 'react';
import { shallow } from 'enzyme';
import InfoColumn from 'components/Form/InfoColumn';

const label = 'Info!';

const button = <button type="button">Submit</button>;

const infoFields = [
  <p key="0">
    <input name="foo" />
  </p>,
];
const metaFields = [
  <p key="0">
    <input name="bar" />
  </p>,
];

describe('InfoColumn', () => {
  test('No props', () => {
    const wrapper = shallow(<InfoColumn />);
    expect(wrapper).toMatchSnapshot();
  });

  test('with infoFields', () => {
    const wrapper = shallow(<InfoColumn infoFields={infoFields} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('with metaFields', () => {
    const wrapper = shallow(<InfoColumn metaFields={metaFields} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('with info and meta fields', () => {
    const wrapper = shallow(<InfoColumn infoFields={infoFields} metaFields={metaFields} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('no button', () => {
    const wrapper = shallow(
      <InfoColumn label={label} infoFields={infoFields} metaFields={metaFields} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('with button', () => {
    const wrapper = shallow(
      <InfoColumn label={label} infoFields={infoFields} metaFields={metaFields} button={button} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
