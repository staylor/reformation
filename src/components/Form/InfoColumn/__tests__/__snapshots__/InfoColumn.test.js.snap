// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoColumn No props 1`] = `
.lights-0 {
  float: right;
  line-height: 1.4;
  margin: 10px -300px 0 0;
  position: -webkit-sticky;
  position: sticky;
  top: 10px;
  width: 280px;
}

@media screen and (max-width:782px) {
  .lights-0 {
    float: none;
    margin-right: 0;
    width: 100%;
  }
}

<section
  className="lights-0"
/>
`;

exports[`InfoColumn no button 1`] = `
.lights-5 {
  float: right;
  line-height: 1.4;
  margin: 10px -300px 0 0;
  position: -webkit-sticky;
  position: sticky;
  top: 10px;
  width: 280px;
}

@media screen and (max-width:782px) {
  .lights-5 {
    float: none;
    margin-right: 0;
    width: 100%;
  }
}

.lights-2 {
  border: 1px solid #e6e6e6;
  box-shadow: 0 1px 1px rgba(0,0,0,0.04);
  box-sizing: border-box;
  display: block;
  margin: 0 0 20px;
}

.lights-0 {
  border-bottom: 1px solid #e6e6e6;
  color: #262626;
  font-size: 14px;
  font-weight: 700;
  padding: 8px 12px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.lights-1 {
  font-size: 13px;
  padding: 6px 10px 20px;
}

<section
  className="lights-5"
>
  <aside
    className="lights-2"
  >
    <h3
      className="lights-0"
    >
      Info!
    </h3>
    <div
      className="lights-1"
    >
      <p
        key="0"
      >
        <input
          name="foo"
        />
      </p>
    </div>
  </aside>
  <aside
    className="lights-2"
  >
    <div
      className="lights-1"
    >
      <p
        key="0"
      >
        <input
          name="bar"
        />
      </p>
    </div>
  </aside>
</section>
`;

exports[`InfoColumn with button 1`] = `
.lights-5 {
  float: right;
  line-height: 1.4;
  margin: 10px -300px 0 0;
  position: -webkit-sticky;
  position: sticky;
  top: 10px;
  width: 280px;
}

@media screen and (max-width:782px) {
  .lights-5 {
    float: none;
    margin-right: 0;
    width: 100%;
  }
}

.lights-2 {
  border: 1px solid #e6e6e6;
  box-shadow: 0 1px 1px rgba(0,0,0,0.04);
  box-sizing: border-box;
  display: block;
  margin: 0 0 20px;
}

.lights-0 {
  border-bottom: 1px solid #e6e6e6;
  color: #262626;
  font-size: 14px;
  font-weight: 700;
  padding: 8px 12px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.lights-1 {
  font-size: 13px;
  padding: 6px 10px 20px;
}

<section
  className="lights-5"
>
  <aside
    className="lights-2"
  >
    <h3
      className="lights-0"
    >
      Info!
    </h3>
    <div
      className="lights-1"
    >
      <p
        key="0"
      >
        <input
          name="foo"
        />
      </p>
      <button
        type="button"
      >
        Submit
      </button>
    </div>
  </aside>
  <aside
    className="lights-2"
  >
    <div
      className="lights-1"
    >
      <p
        key="0"
      >
        <input
          name="bar"
        />
      </p>
    </div>
  </aside>
</section>
`;

exports[`InfoColumn with info and meta fields 1`] = `
.lights-5 {
  float: right;
  line-height: 1.4;
  margin: 10px -300px 0 0;
  position: -webkit-sticky;
  position: sticky;
  top: 10px;
  width: 280px;
}

@media screen and (max-width:782px) {
  .lights-5 {
    float: none;
    margin-right: 0;
    width: 100%;
  }
}

.lights-2 {
  border: 1px solid #e6e6e6;
  box-shadow: 0 1px 1px rgba(0,0,0,0.04);
  box-sizing: border-box;
  display: block;
  margin: 0 0 20px;
}

.lights-0 {
  border-bottom: 1px solid #e6e6e6;
  color: #262626;
  font-size: 14px;
  font-weight: 700;
  padding: 8px 12px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.lights-1 {
  font-size: 13px;
  padding: 6px 10px 20px;
}

<section
  className="lights-5"
>
  <aside
    className="lights-2"
  >
    <h3
      className="lights-0"
    />
    <div
      className="lights-1"
    >
      <p
        key="0"
      >
        <input
          name="foo"
        />
      </p>
    </div>
  </aside>
  <aside
    className="lights-2"
  >
    <div
      className="lights-1"
    >
      <p
        key="0"
      >
        <input
          name="bar"
        />
      </p>
    </div>
  </aside>
</section>
`;

exports[`InfoColumn with infoFields 1`] = `
.lights-3 {
  float: right;
  line-height: 1.4;
  margin: 10px -300px 0 0;
  position: -webkit-sticky;
  position: sticky;
  top: 10px;
  width: 280px;
}

@media screen and (max-width:782px) {
  .lights-3 {
    float: none;
    margin-right: 0;
    width: 100%;
  }
}

.lights-2 {
  border: 1px solid #e6e6e6;
  box-shadow: 0 1px 1px rgba(0,0,0,0.04);
  box-sizing: border-box;
  display: block;
  margin: 0 0 20px;
}

.lights-0 {
  border-bottom: 1px solid #e6e6e6;
  color: #262626;
  font-size: 14px;
  font-weight: 700;
  padding: 8px 12px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.lights-1 {
  font-size: 13px;
  padding: 6px 10px 20px;
}

<section
  className="lights-3"
>
  <aside
    className="lights-2"
  >
    <h3
      className="lights-0"
    />
    <div
      className="lights-1"
    >
      <p
        key="0"
      >
        <input
          name="foo"
        />
      </p>
    </div>
  </aside>
</section>
`;

exports[`InfoColumn with metaFields 1`] = `
.lights-2 {
  float: right;
  line-height: 1.4;
  margin: 10px -300px 0 0;
  position: -webkit-sticky;
  position: sticky;
  top: 10px;
  width: 280px;
}

@media screen and (max-width:782px) {
  .lights-2 {
    float: none;
    margin-right: 0;
    width: 100%;
  }
}

.lights-1 {
  border: 1px solid #e6e6e6;
  box-shadow: 0 1px 1px rgba(0,0,0,0.04);
  box-sizing: border-box;
  display: block;
  margin: 0 0 20px;
}

.lights-0 {
  font-size: 13px;
  padding: 6px 10px 20px;
}

<section
  className="lights-2"
>
  <aside
    className="lights-1"
  >
    <div
      className="lights-0"
    >
      <p
        key="0"
      >
        <input
          name="bar"
        />
      </p>
    </div>
  </aside>
</section>
`;
